packages/sanity/src/core/form/inputs/arrays/ArrayOfObjectsInput/List/ListArrayInput.tsx

```diff
@@ -0,0 +1,103 @@
+/* eslint-disable react/jsx-handler-names */
+import {Card, Stack, Text} from '@sanity/ui'
+import React, {useCallback} from 'react'
+import {UploaderResolver} from '../../../../studio/uploads/types'
+import {Item, List} from '../../common/list'
+import {ArrayOfObjectsInputProps, ObjectItem, UploadEvent} from '../../../../types'
+import {DefaultArrayInputFunctions} from '../../common/ArrayFunctions'
+import {ArrayOfObjectsItem} from '../../../../members'
+
+import {createProtoArrayValue} from '../createProtoArrayValue'
+import {UploadTargetCard} from '../../common/UploadTargetCard'
+import {ErrorItem} from './ErrorItem'
+
+export interface ArrayInputProps<Item extends ObjectItem> extends ArrayOfObjectsInputProps<Item> {
+  resolveUploader: UploaderResolver
+  onUpload: (event: UploadEvent) => void
+}
+
+export function ListArrayInput<Item extends ObjectItem>(props: ArrayInputProps<Item>) {
+  const {
+    schemaType,
+    onChange,
+    value = [],
+    readOnly,
+    members,
+    elementProps,
+    resolveUploader,
+    onInsert,
+    onItemMove,
+    onUpload,
+    renderPreview,
+    renderField,
+    renderItem,
+    renderInput,
+  } = props
+
+  const handlePrepend = useCallback(
```

Comment:
Consider moving these two to a reusable `useArrayHandlers` or something, that takes `ArrayInputProps` and returns `{handlePrepend, handleAppend}` (other things?) and share with GridArrayInput